phpBB/phpbb/extension/base.php

```diff
@@ -137,6 +139,30 @@ protected function get_migration_file_list()
 
 		$migrations = $this->extension_finder->get_classes_from_files($migrations);
 
+		if ($validate_classes)
+		{
+			foreach ($migrations as $key => $migration)
+			{
+				// If the class exists and is a subclass of the
+				// \phpbb\db\migration\migration abstract class
+				// we skip it.
+
+				// Otherwise, i.e. if it doesn't exist or it is
+				// not an extend the abstract class, we unset it
+				if (class_exists($migration))
+				{
+					$reflector = new \ReflectionClass($migration);
+					if ($reflector->isSubclassOf('\phpbb\db\migration\migration'))
```

Comment:
I think this needs to be `if ($reflection->IsInstantiable())`
Otherwise files that still extend the migration class but are meant to be extended themselves (like container_aware_migration) will still be run when they shouldn't.